inc/class-ubc-isp-theme-functions.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * This class is responsible for theme functions
+ *
+ * @package UBC Collab ISP Child Theme Functions
+ */
+class UBC_ISP_Theme_Functions {
+
+	/**
+	 * Initialize the class
+	 */
+	public static function init() {
+		self::theme_hooks();
+	}
+
+	/**
+	 * Setup the theme hooks
+	 */
+	public static function theme_hooks() {   
+		add_action( 'after_setup_theme', array( 'UBC_ISP_Theme_Functions', 'theme_setup' ), 20, 2 );
+
+		add_action( 'wp_enqueue_scripts', array( 'UBC_ISP_Theme_Functions', 'theme_scripts' ) );
+		add_action( 'enqueue_block_editor_assets', array( 'UBC_ISP_Theme_Functions', 'editor_scripts' ) );
+
+		add_action( 'wp-hybrid-clf_before_header', array( 'UBC_ISP_Theme_Functions', 'open_header_wrap' ) );
+		add_action( 'wp-hybrid-clf_after_header', array( 'UBC_ISP_Theme_Functions', 'close_header_wrap' ) );
+
+		add_action( 'wp-hybrid-clf_before_footer', array( 'UBC_ISP_Theme_Functions', 'open_footer_wrap' ) );
+		add_action( 'wp-hybrid-clf_after_footer', array( 'UBC_ISP_Theme_Functions', 'close_footer_wrap' ) );
+	}
+
+	/**
+	 * Enqueue scripts and styles.
+	 */
+	public static function theme_scripts() {
+
+		$theme = wp_get_theme();
+		wp_enqueue_style( 'style', get_template_directory_uri() . '/style.css', array(), $theme->parent()->get( 'Version' ) );
+	   
+		wp_enqueue_style( 'dataTable-style', '//cdn.datatables.net/1.13.4/css/jquery.dataTables.min.css', array(), CHILD_THEME_VERSION );
```

Comment:
Is it possible we can enqueue these only when needed? Rather than on every page load?

Reply:
I hadn't realized TablePress was in use on the site and that loads DataTables by default so that script was not needed. I only ended up using slick carousel in one spot so I rewrote that to remove slick and remove the enqueue as well.